client/components/notice/index.tsx

```diff
@@ -46,132 +46,91 @@ interface NoticeProps {
-		}
-	}
-
-	getIcon(): string {
-		let icon: string;
-
-		switch ( this.props.status ) {
-			case 'is-info':
-				icon = 'info';
-				break;
-			case 'is-success':
-				icon = 'checkmark';
-				break;
-			case 'is-error':
-				icon = 'notice';
-				break;
-			case 'is-warning':
-			case 'is-transparent-info':
-				icon = 'notice';
-				break;
-			default:
-				icon = 'info';
-				break;
-		}
-
-		return icon;
+function getIcon( status: NoticeStatus | undefined ): string {
+	switch ( status ) {
+		case 'is-success':
+			return 'checkmark';
+		case 'is-error':
+		case 'is-warning':
+		case 'is-transparent-info':
+			return 'notice';
+		case 'is-info':
+		default:
+			return 'info';
 	}
+}
 
-	render(): ReactNode {
-		const {
-			children,
-			className,
-			icon,
-			isCompact,
-			isLoading,
-			onDismissClick,
-			showDismiss = ! isCompact,
-			status,
-			text,
-			translate,
-			theme = 'dark',
-		} = this.props;
-
-		const classes = clsx( 'notice', status, className, {
-			'is-compact': isCompact,
-			'is-loading': isLoading,
-			'is-dismissable': showDismiss,
-			/**
-			 * The default scss styling of this component is of theme dark
-			 * and we only have 1 override theme which is the light theme
-			 */
-			'is-light': theme === 'light',
-		} );
-
-		let iconNeedsDrop = false;
-		let renderedIcon: ReactNode = null;
-
-		if ( icon && isValidElement( icon ) ) {
-			renderedIcon = icon;
-		} else {
-			const iconName = icon || this.getIcon();
-			renderedIcon = <Gridicon className="notice__icon" icon={ iconName as string } size={ 24 } />;
-			iconNeedsDrop = GRIDICONS_WITH_DROP.includes(
-				iconName as ( typeof GRIDICONS_WITH_DROP )[ number ]
-			);
-		}
-
-		return (
-			<div className={ classes } role="status" aria-label={ translate( 'Notice' ) }>
-				<span className="notice__icon-wrapper">
-					{ iconNeedsDrop && <span className="notice__icon-wrapper-drop" /> }
-					{ renderedIcon }
-				</span>
-				<span className="notice__content">
-					<span className="notice__text">{ text ? text : children }</span>
-				</span>
-				{ text ? children : null }
-				{ showDismiss && (
-					<button
-						className="notice__dismiss"
-						onClick={ onDismissClick }
-						aria-label={ translate( 'Dismiss' ) }
-					>
-						<Gridicon icon="cross" size={ 24 } />
-					</button>
-				) }
-			</div>
+export default function Notice( {
+	children,
+	className,
+	duration = 0,
+	icon,
+	isCompact = false,
+	isLoading = false,
+	onDismissClick = noop,
+	showDismiss = ! isCompact,
+	status,
+	text,
+	theme = 'dark',
+}: NoticeProps ) {
+	const translate = useTranslate();
+
+	const onDismissClickRef = useRef( onDismissClick );
+	useEffect( () => {
+		onDismissClickRef.current = onDismissClick;
+	}, [ onDismissClick ] );
+
+	useEffect( () => {
+		const dismissTimeout = setTimeout( onDismissClickRef.current, duration );
+		return () => clearTimeout( dismissTimeout );
+	}, [ duration ] );
```

Comment:
If  the `duration` prop is not passed, it defaults to `0`, which means the notice is immediately hidden. Previously, we were only calling `setTimeout()` if `this.props.duration! > 0`. I'm guessing we want to keep that behaviour here?

Reply:
Agreed, I don't see a reason not to keep this behavior

Reply:
Fixed that. It broke one of unit tests 🙂 While debugging that test I got to look at the `GlobalNotices` component, and it was super-easy to also refactor. So this PR refactors both `Notice` and `GlobalNotices`.